wemake_python_styleguide/visitors/ast/blocks.py

```diff
@@ -255,7 +255,8 @@ def _add_to_scope(self, node: ast.AST, names: Set[str]) -> None:
     def _check_variable_usage(self, node: ast.Name) -> None:
         context = cast(ast.AST, get_context(node))
         blocks = self._block_variables[context][node.id]
-        if all(is_contained_by(node, block) for block in blocks):
+
+        if not blocks or any(is_contained_by(node, block) for block in blocks):
```

Comment:
Let's create a new variables out of this condition. I am a bit lost reading it. 
We need some extra meta context.

Reply:
Added a new variable and some context to understand it.